connection.go

```diff
@@ -8,17 +8,41 @@ import (
 	amqp "github.com/rabbitmq/amqp091-go"
 )
 
-// ErrUnexpectedConnClosed is returned by ListenAndServe() if the server
-// shuts down without calling Stop() and if AMQP does not give an error
-// when said shutdown happens.
-var ErrUnexpectedConnClosed = errors.New("unexpected connection close without specific error")
+var (
+	// ErrUnexpectedConnClosed happens when the [Server] or [Client] gets a
+	// connection closed without a specific error from amqp091. This usually
+	// happens if the user calls Close() on a amqp connection or channel. You
+	// can check for this error using [Server.OnError] or [Client.OnError].
```

Comment:
nit/personal: Don't think we should document where this error can be checked, I'd rather just document close to `OnError` and refer to the public errors for expected error messages.

Reply:
Will fix!